src/_pytest/pathlib.py

```diff
@@ -46,38 +47,53 @@ def ensure_reset_dir(path):
     path.mkdir()
 
 
-def rm_rf(path):
-    """Remove the path contents recursively, even if some elements
-    are read-only.
-    """
+def on_rm_rf_error(func, path, exc, **kwargs):
+    """Handles known read-only errors during rmtree."""
+    start_path = kwargs["start_path"]
```

Comment:
should this be `kwargs.pop(...)` and then `if not kwargs: raise TypeError(...)` such that we retain the api from 5.x?

Reply:
Good point, but given that this is called by Python's stdlib I don'think it actually matters.

Reply:
ok, as long as it's not an api we could ever expose to users 👍